api/src/shared/infrastructure/repositories/assessment-repository.js

```diff
@@ -32,36 +30,28 @@ const get = async function (id, domainTransaction = DomainTransaction.emptyTrans
   return new Assessment(assessment);
 };
 
-const findLastCompletedAssessmentsForEachCompetenceByUser = function (userId, limitDate) {
-  return BookshelfAssessment.collection()
-    .query((qb) => {
-      qb.join('assessment-results', 'assessment-results.assessmentId', 'assessments.id');
-      qb.where({ userId })
-        .where(function () {
-          this.where({ type: 'PLACEMENT' });
-        })
-        .where('assessments.createdAt', '<', limitDate)
-        .where('assessment-results.createdAt', '<', limitDate)
-        .where('assessments.state', '=', 'completed')
-        .orderBy('assessments.createdAt', 'desc');
+const findLastCompletedAssessmentsForEachCompetenceByUser = async function (userId, limitDate) {
+  const lastCompletedAssessments = await knex('assessments')
+    .select('assessments.*')
+    .join('assessment-results', 'assessment-results.assessmentId', 'assessments.id')
+    .where({ userId })
+    .where(function () {
+      this.where({ type: 'PLACEMENT' });
     })
-    .fetch({ require: false })
-    .then((bookshelfAssessmentCollection) => bookshelfAssessmentCollection.models)
-    .then(_selectLastAssessmentForEachCompetence)
-    .then((assessments) => bookshelfToDomainConverter.buildDomainObjects(BookshelfAssessment, assessments));
-};
+    .where('assessments.createdAt', '<', limitDate)
+    .where('assessment-results.createdAt', '<', limitDate)
+    .where('assessments.state', '=', 'completed')
+    .orderBy('assessments.createdAt', 'desc');
 
-const getByAssessmentIdAndUserId = function (assessmentId, userId) {
-  return BookshelfAssessment.query({ where: { id: assessmentId }, andWhere: { userId } })
-    .fetch()
-    .then((assessment) => bookshelfToDomainConverter.buildDomainObject(BookshelfAssessment, assessment))
-    .catch((error) => {
-      if (error instanceof BookshelfAssessment.NotFoundError) {
-        throw new NotFoundError();
-      }
+  return _selectLastAssessmentForEachCompetence(lastCompletedAssessments).map(
+    (assessment) => new Assessment(assessment),
+  );
+};
```

Comment:
Je propose d'ajouter plus de contexte pour les noms de colonnes, et simplifier la condition du type d'assessment :
```suggestion
const findLastCompletedAssessmentsForEachCompetenceByUser = async function (userId, limitDate) {
  const lastCompletedAssessments = await knex('assessments')
    .select('assessments.*')
    .join('assessment-results', 'assessment-results.assessmentId', 'assessments.id')
    .where({ 'assessments.userId': userId })
    .where({ 'assessments.type': 'PLACEMENT' })
    .where('assessments.createdAt', '<', limitDate)
    .where('assessment-results.createdAt', '<', limitDate)
    .where('assessments.state', '=', 'completed')
    .orderBy('assessments.createdAt', 'desc');

  return _selectLastAssessmentForEachCompetence(lastCompletedAssessments).map(
    (assessment) => new Assessment(assessment),
  );
};
```